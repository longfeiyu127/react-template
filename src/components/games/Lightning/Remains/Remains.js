import React from 'react';
// import Tictactoe from './tictactoe'
// import Fivechess from './fivechess'
import './Remains.less';

class Remains extends React.Component {
  constructor() {
    super();
    this.state = {};
  }

  componentDidMount() {
    // eslint-disable-next-line
    const { index } = this.props;
    setTimeout(() => {
      // eslint-disable-next-line
      this.props.delRemains(index);
    }, 1550);
  }

  render() {
    // eslint-disable-next-line
    const { position, size } = this.props;
    return (
      <div
        className="g-lig-remains"
        style={{
          top: `${position.top}rem`,
          left: `${position.left}rem`,
          height: `${size.height}rem`,
          width: `${size.width}rem`
        }}
      >
        {/* style={{fontWeight: (move === stepNumber) ? 900 : 500, cursor: 'pointer'}} */}
        <main className="remains-main" />
      </div>
    );
  }
}

export default Remains;
