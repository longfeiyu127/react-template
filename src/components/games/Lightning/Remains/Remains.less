.g-lig-remains{
  position: absolute;
  >.remains-main {
    height: 100%;
    width: 100%;
    animation: switchover 1500ms linear;
  }
  // >.player{
  //   background: url('../../../../../assets/images/games/lightning/bullet/palyerBullet.png') no-repeat center/100% 100%;
  //   animation: flicker 100ms infinite;
  // }
  // >.enemyA0{
  //   background: url('../../../../../assets/images/games/lightning/enemy/enemy_typeA.png') no-repeat center/100% 100%;
  // }
  // >.enemyA1{
  //   background: url('../../../../../assets/images/games/lightning/enemy/enemy_typeA1.png') no-repeat center/100% 100%;
  // }
  // >.enemyA2{
  //   background: url('../../../../../assets/images/games/lightning/enemy/enemy_typeA2.png') no-repeat center/100% 100%;
  // }
  // >.enemyC{
  //   background: url('../../../../../assets/images/games/lightning/enemy/enemy_typeC.png') no-repeat center/100% 100%;
  // }

  // >.explosion{
  //   background: url('../../../../../assets/images/games/lightning/explosion/disp_02.png') no-repeat center/100% 100%;
  // }
  @keyframes switchover {
    0% {
      background: url('../../../../assets/images/games/lightning/explosion/disp_01.png') no-repeat center/100% 100%;
    }
    20% {
      background: url('../../../../assets/images/games/lightning/explosion/disp_02.png') no-repeat center/100% 100%;
    }
    40% {
      background: url('../../../../assets/images/games/lightning/explosion/disp_03.png') no-repeat center/100% 100%;
    }
    60% {
      background: url('../../../../assets/images/games/lightning/explosion/disp_04.png') no-repeat center/100% 100%;
    }
    80% {
      background: url('../../../../assets/images/games/lightning/explosion/disp_05.png') no-repeat center/100% 100%;
    }
    95% {
      background: url('../../../../assets/images/games/lightning/explosion/disp_06.png') no-repeat center/100% 100%;
    }
    100% {
      background: url('../../../../assets/images/games/lightning/explosion/disp_06.png') no-repeat center/100% 100%;
      opacity: 0.5;
    }
  }
}
