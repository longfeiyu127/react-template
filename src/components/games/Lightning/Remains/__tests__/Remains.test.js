import React from 'react';
import renderer from 'react-test-renderer';
// import Enzyme, { mount } from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
import Remains from '../Remains';

// Enzyme.configure({ adapter: new Adapter() });

describe('<Remains />', () => {
  it('component render', () => {
    const position = {
      top: 0.1,
      left: 0.1
    };
    const size = {
      height: 0.1,
      width: 0.1
    };
    let remains = renderer
      .create(
        <Remains
          position={position}
          size={size}
          index={1}
          delRemains={() => {}}
        />
      )
      .toJSON();
    expect(remains).toMatchSnapshot();

    remains = renderer
      .create(
        <Remains
          position={position}
          size={size}
          index={1}
          delRemains={() => {}}
        />
      )
      .toJSON();
    expect(remains).toMatchSnapshot();
  });

  // it('componentDidMount', () => {
  //   const remains = new Remains();
  //   console.log(remains.componentDidMount);
  //   // const fn = jest.fn();
  //   // const position = {
  //   //   top: 0.1,
  //   //   left: 0.1
  //   // };
  //   // const size = {
  //   //   height: 0.1,
  //   //   width: 0.1
  //   // };
  //   // const remains = mount(
  //   //   <Remains
  //   //     position={position}
  //   //     size={size}
  //   //     index={1}
  //   //     delRemains={fn}
  //   //   />
  //   // );
  //   // remains.instance().componentDidMount();
  //   // expect(fn.mock.calls.length).toBe(1);
  // });
});
