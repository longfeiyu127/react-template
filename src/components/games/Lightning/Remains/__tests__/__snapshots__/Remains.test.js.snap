// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Remains /> component render 1`] = `
<div
  className="g-lig-remains"
  style={
    Object {
      "height": "0.1rem",
      "left": "0.1rem",
      "top": "0.1rem",
      "width": "0.1rem",
    }
  }
>
  <main
    className="remains-main"
  />
</div>
`;

exports[`<Remains /> component render 2`] = `
<div
  className="g-lig-remains"
  style={
    Object {
      "height": "0.1rem",
      "left": "0.1rem",
      "top": "0.1rem",
      "width": "0.1rem",
    }
  }
>
  <main
    className="remains-main"
  />
</div>
`;
